Plugins/DependencyPlugin/ProjectDescriptionHelpers/ModuleType.swift

```diff
@@ -0,0 +1,86 @@
+//
+//  ModuleType.swift
+//  Plugins
+//
+//  Created by Seoyeon Choi on 11/10/25.
+//
+
+import Foundation
+
+public enum ModuleType {
+    case feature(Feature)
+    case domain(Domain)
+    case data(Data)
+    case core(Core)
+    case ui(UI)
+}
+
+public extension ModuleType {
+    enum Feature: String {
+        case HomeFeature
+        case FeedFeature
+        
+        public var name: String { self.rawValue }
+
+        func targetName(type: TargetType) -> String {
+            "\(self.rawValue)\(type.rawValue)"
+        }
+    }
+}
```

Comment:
W2

큰건 아닌데, 이번에 코드 분석을 하다가 제가 잘 모르는 내용이라 코드를 계속 들여다 보니 조금 수정 되면 가독성이 더 좋겠다..! 싶은 것이 있어서 제안해봅니다.

현재 targetName에서 rawValue를 통해 조합하고 있는데, 그러면 이 targetName이라는 메서드의 역할을 알기 위해서 각각의 rawValue가 뭐지? 를 찾아보게 되더라구요. 다 보고 나니까 이해는 되는데, 보다 직관적으로 이해할 수 있게 & case이름이 보다 case이름  답도록 하기 위해서
아래와 같이 수정하면 어떨까요?

++ 문자열을 안전하게 쓰기 위해서 enum을 도입한건데, enum 의 케이스 명이 바뀌면 결과적으로 name프로퍼티 값이 바뀌거나, 함수 내에서 사용하는 값도 바뀌는 거라 결국 문자열 하드코딩과 비슷해진 느낌이 듭니다!



```suggestion
enum ModuleType {
    enum Feature {
        case home
        case feed

        var name: String {
            switch self {
            case .home: return "HomeFeature"
            case .feed: return "FeedFeature"
            }
        }

        func targetName(type: TargetType) -> String {
            name + type.suffix
        }
    }
}

```

++ TargetType 수정 제안.
이 역시 rawValue보다 suffix로 쓰이는 외부 네이밍 규칙을 표현할 수 있는 프로퍼티로 추가하는건 어떨까요?

```swift
public enum TargetType {
    case sources
    case demo
    case tests

    var suffix: String {
        switch self {
        case .sources: return ""
        case .demo: return "Demo"
        case .tests: return "Tests"
        }
    }
}

```

Reply:
좋습니다..! rawValue를 사용하는 것보다 훨씬 가독성이 높아져서 좋네여!!

Reply:
131a7754e0c0ea95b7319b4d4e36ad0cac0cce7f